python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py

```diff
@@ -0,0 +1,388 @@
+            Tensor of source node features. Shape: (num_src_nodes, in_feat).
+        coeff : torch.Tensor, dtype=torch.float32, requires_grad=True
+            The coefficient matrix used in basis-decomposition regularization.
+            Shape: (num_rels, num_bases). It should be set to ``None``
+            when ``regularizer=None``.
+
+        Returns
+        -------
+        agg_output : torch.Tensor, dtype=torch.float32
+            Aggregation output. Shape: (num_dst_nodes, num_rels * in_feat)
+            when ``regularizer=None``; Shape: (num_dst_nodes, num_bases * in_feat)
+            when ``regularizer='basis'``.
+        """
+        ctx.device = g.device
+        if g.idtype == th.int32:
+            mfg_csr_func = message_flow_graph_hg_csr_int32
+            agg_fwd_func = agg_hg_basis_post_fwd_int32
+        elif g.idtype == th.int64:
+            mfg_csr_func = message_flow_graph_hg_csr_int64
+            agg_fwd_func = agg_hg_basis_post_fwd_int64
+        else:
+            raise TypeError(
+                f"Supported ID type: torch.int32 or torch.int64, "
+                f"but got {g.idtype}"
+            )
+        ctx.idtype = g.idtype
+
+        _in_feat = feat.shape[-1]
+        indptr, indices, edge_ids = g.adj_sparse("csc")
+        # edge_ids is in a mixed order, need to permutate incoming etypes
+        # and stash the result for backward propagation
+        ctx.edge_types_int32 = edge_types[edge_ids.long()].int()
+        # node_types are not needed by the post-variant RGCN aggregation functions
+        _num_node_types = 0
+        _out_node_types = _in_node_types = None
+
+        mfg = mfg_csr_func(
+            fanout,
+            g.dstnodes(),
+            g.srcnodes(),
+            indptr,
+            indices,
+            _num_node_types,
+            num_rels,
+            _out_node_types,
+            _in_node_types,
+            ctx.edge_types_int32,
+        )
+        ctx.mfg = mfg
+
+        if coeff is None:
+            leading_dimension = num_rels * _in_feat
+        else:
+            _num_bases = coeff.shape[-1]
+            leading_dimension = _num_bases * _in_feat
+
+        agg_output = th.empty(
+            g.num_dst_nodes(),
+            leading_dimension,
+            dtype=th.float32,
+            device=ctx.device,
+        )
+        if coeff is None:
+            agg_fwd_func(agg_output, feat.detach(), mfg)
+        else:
+            agg_fwd_func(
+                agg_output,
+                feat.detach(),
+                mfg,
+                weights_combination=coeff.detach()
+            )
+
+        ctx.save_for_backward(feat, coeff)
+        return agg_output
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        r"""Compute the backward pass of R-GCN aggregation layer.
+
+        Parameters
+        ----------
+        grad_output : torch.Tensor, dtype=torch.float32
+            Gradient of loss function w.r.t output.
```

Comment:
Ditto for `ctx` and removing dtype of `grad_output`.

Reply:
dtype removed. Regarding ctx, see comments above.